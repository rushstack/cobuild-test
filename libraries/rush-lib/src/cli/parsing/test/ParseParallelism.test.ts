// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { parseParallelism } from '../ParseParallelism';

describe(parseParallelism.name, () => {
  it('throwsErrorOnInvalidParallelism', () => {
    expect(() => parseParallelism('tequila')).toThrowErrorMatchingSnapshot();
  });

  it('createsWithPercentageBasedParallelism', () => {
    const value: number = parseParallelism('50%', 20);
    expect(value).toEqual(10);
  });

  it('throwsErrorOnInvalidParallelismPercentage', () => {
    expect(() => parseParallelism('200%')).toThrowErrorMatchingSnapshot();
  });
});
