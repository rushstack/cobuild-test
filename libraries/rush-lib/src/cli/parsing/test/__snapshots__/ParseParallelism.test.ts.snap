// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseParallelism throwsErrorOnInvalidParallelism 1`] = `"Invalid parallelism value of 'tequila', expected a number, a percentage, or 'max'"`;

exports[`parseParallelism throwsErrorOnInvalidParallelismPercentage 1`] = `"Invalid percentage value of '200%', value cannot be less than '0%' or more than '100%'"`;
